drivers: ieee802154: make drivers DT-based

gmarull opened the pull request:
Many IEEE 802.15.4 drivers were still using DEVICE_DEFINE and Kconfig names, this patch converts all of them to be DT-based. Note that in some cases I've guessed the DT representation from existing code or datasheets, but it may not be the best choice.

Part of https://github.com/zephyrproject-rtos/zephyr/pull/47938 (only driver conversion here).

NOTE: `label` property is kept because it is still required due to `NET_CONFIG_IEEE802154_DEV_NAME`. See #47938 for next steps towards its removal.

gmarull committed:
drivers: ieee802154: cc13x2_cc26x2: make driver DT-based

Use Devicetree to describe the radio and IEEE 802.15.4. This allows to
remove usage of IEEE802154_CC13XX_CC26XX_DRV_NAME in preparation for the
removal of NET_CONFIG_IEEE802154_DEV_NAME. All boards used in testing
have been updated to enable the peripheral in DT as well.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
drivers: ieee802154: rf2xx: remove unused IEEE802154_RF2XX_DRV_NAME

The option is no longer used (driver is DT-based).

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
drivers: ieee802154: nrf5: make driver DT-based

Use Devicetree to describe the radio and IEEE 802.15.4. This allows to
remove usage of IEEE802154_NRF5_DRV_NAME in preparation for the
removal of NET_CONFIG_IEEE802154_DEV_NAME.

All SoC files have been updated with the addition of an ieee802154 node
(disabled and only on those SoCs that define ieee802154-supported. The
peripheral has been enabled in the nRF52840DK board (used for testing
ieee802154).

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
drivers: ieee802154: kw41z: make driver DT-based

Use Devicetree to describe the radio and IEEE 802.15.4. This allows to
remove usage of IEEE802154_KW41Z_DRV_NAME in preparation for the removal
of NET_CONFIG_IEEE802154_DEV_NAME.

KW41Z files have been updated with the addition of radio and an
ieee802154 nodes The peripheral has been enabled in the frdm_k41z board
(used for testing ieee802154).

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
drivers: ieee802154: cc1200: make driver DT-based

Use Devicetree to describe the radio and IEEE 802.15.4. This allows to
remove usage of IEEE802154_CC1200_DRV_NAME in preparation for the removal
of NET_CONFIG_IEEE802154_DEV_NAME.

In this case, the driver already had bindings, however, it was still
using NET_DEVICE_INIT.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
drivers: ieee802154: cc13x2_cc26x2_subg: make driver DT-based

Use Devicetree to describe the radio and IEEE 802.15.4. This allows to
remove usage of IEEE802154_CC13XX_CC26XX_SUBG_DRV_NAME in preparation
for the removal of NET_CONFIG_IEEE802154_DEV_NAME.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
drivers: ieee802154: uart_pipe: split Kconfig

As all other drivers, keep uart_pipe driver Kconfig settings into its
own file.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
drivers: ieee802154: uart_pipe: make driver DT-based

Use Devicetree to describe the UART UPIPE IEEE 802.15.4 driver. This
allows to remove usage of IEEE802154_UPIPE_DRV_NAME in preparation for
the removal of NET_CONFIG_IEEE802154_DEV_NAME.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

galak commented on dts/bindings/crypto/ti,cc2520-crypto.yaml:
```diff
@@ -0,0 +1,8 @@
+
+description: TI CC2520 Cryptographic Accelerator
+
+compatible: "ti,cc2520-crypto"
```
I don't think we should add this node, we are doing this because of how we structured the driver/etc.

I think we should add a `struct device` in `ieee802154_config` that sets an optional pointer to a crypto device.  If that isn't set than `ieee802154_security_init` will fall back to a software option.

This way the CC2520 can keep the DEVICE_DEFINE() and the `struct device` will end up being local to the driver.

gmarull replied:
fair point, just dropped CONFIG_IEEE802154_CC2520_CRYPTO_DRV_NAME for now. please check

gmarull committed:
drivers: ieee802154: cc2520: drop IEEE802154_CC2520_CRYPTO_DRV_NAME

The crypto driver is used internally, so there's no real need to expose
its name as a Kconfig option. Just drop it in favor of a plain string
with the same previous value.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

galak commented on dts/bindings/ieee802154/ti,cc13xx-cc26xx-ieee802154-subghz.yaml:
```diff
@@ -0,0 +1,8 @@
+
+description: TI SimpleLink CC13xx / CC26xx IEEE 802.15.4 node (sub-GHz)
+
+compatible: "ti,cc13xx-cc26xx-ieee802154-subghz"
```
@cfriedt is subghz really just a mode?  Seems like its not a different device?  Can you really have both subghz and non-subghz drivers enabled at the same time?

galak commented on dts/arm/ti/cc13x2_cc26x2.dtsi:
```diff
@@ -128,6 +128,28 @@
+			       0x40041000 0x2000   /* RFC_DBELL */
+			       0x40043000 0x1000   /* RFC_RAT */
+			       0x40044000 0x1000>; /* RFC_FSCA */
+
```
probably good to add reg-names

gmarull replied:
added

cfriedt reviewed (approved):
LGTM

galak commented:
Hmm, I guess I didn't mark this as changes needed and it got merged.  Oh well...

gmarull commented:
> Hmm, I guess I didn't mark this as changes needed and it got merged. Oh well...

which change was pending?

galak commented:
> which change was pending?

I made a comment about the crypto devicetree node.  Its not critical.

gmarull commented:
> > which change was pending?
> 
> I made a comment about the crypto devicetree node. Its not critical.

it was removed, it's now a non-DT device.
